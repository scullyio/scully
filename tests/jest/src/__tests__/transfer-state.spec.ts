import { extractTransferState, readPage } from '../test-config.helper';
import { expect } from '@jest/globals';

describe('TransferState', () => {
  it('should add state to page 1', () => {
    const index: string = readPage('user/1');
    const pageTransferState = extractTransferState(index);
    expect(pageTransferState.user).toMatchSnapshot();
    expect(pageTransferState.posts).toMatchSnapshot();
  });

  it('should add state to page 2', () => {
    const index: string = readPage('user/1/post/1');
    const pageTransferState = extractTransferState(index);
    // expect(cleanIndex).toMatchSnapshot();
    expect(pageTransferState.posts).toMatchSnapshot();
    expect(pageTransferState.user).toMatchSnapshot();
  });
});
