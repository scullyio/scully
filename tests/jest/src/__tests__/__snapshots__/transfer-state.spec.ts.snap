// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransferState should add state to page 1 1`] = `
Object {
  "address": Object {
    "city": "Gwenborough",
    "geo": Object {
      "lat": "-37.3159",
      "lng": "81.1496",
    },
    "street": "Kulas Light",
    "suite": "Apt. 556",
    "zipcode": "92998-3874",
  },
  "company": Object {
    "bs": "harness real-time e-markets",
    "catchPhrase": "Multi-layered client-server neural-net",
    "name": "Romaguera-Crona",
  },
  "email": "Sincere@april.biz",
  "id": 1,
  "name": "Leanne Graham",
  "phone": "1-770-736-8031 x56442",
  "username": "Bret",
  "website": "hildegard.org",
}
`;

exports[`TransferState should add state to page 1 2`] = `
Array [
  Object {
    "body": "quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto",
    "id": 1,
    "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
    "userId": 1,
  },
  Object {
    "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
    "id": 2,
    "title": "qui est esse",
    "userId": 1,
  },
  Object {
    "body": "et iusto sed quo iure
voluptatem occaecati omnis eligendi aut ad
voluptatem doloribus vel accusantium quis pariatur
molestiae porro eius odio et labore et velit aut",
    "id": 3,
    "title": "ea molestias quasi exercitationem repellat qui ipsa sit aut",
    "userId": 1,
  },
  Object {
    "body": "ullam et saepe reiciendis voluptatem adipisci
sit amet autem assumenda provident rerum culpa
quis hic commodi nesciunt rem tenetur doloremque ipsam iure
quis sunt voluptatem rerum illo velit",
    "id": 4,
    "title": "eum et est occaecati",
    "userId": 1,
  },
  Object {
    "body": "repudiandae veniam quaerat sunt sed
alias aut fugiat sit autem sed est
voluptatem omnis possimus esse voluptatibus quis
est aut tenetur dolor neque",
    "id": 5,
    "title": "nesciunt quas odio",
    "userId": 1,
  },
  Object {
    "body": "ut aspernatur corporis harum nihil quis provident sequi
mollitia nobis aliquid molestiae
perspiciatis et ea nemo ab reprehenderit accusantium quas
voluptate dolores velit et doloremque molestiae",
    "id": 6,
    "title": "dolorem eum magni eos aperiam quia",
    "userId": 1,
  },
  Object {
    "body": "dolore placeat quibusdam ea quo vitae
magni quis enim qui quis quo nemo aut saepe
quidem repellat excepturi ut quia
sunt ut sequi eos ea sed quas",
    "id": 7,
    "title": "magnam facilis autem",
    "userId": 1,
  },
  Object {
    "body": "dignissimos aperiam dolorem qui eum
facilis quibusdam animi sint suscipit qui sint possimus cum
quaerat magni maiores excepturi
ipsam ut commodi dolor voluptatum modi aut vitae",
    "id": 8,
    "title": "dolorem dolore est ipsam",
    "userId": 1,
  },
  Object {
    "body": "consectetur animi nesciunt iure dolore
enim quia ad
veniam autem ut quam aut nobis
et est aut quod aut provident voluptas autem voluptas",
    "id": 9,
    "title": "nesciunt iure omnis dolorem tempora et accusantium",
    "userId": 1,
  },
  Object {
    "body": "quo et expedita modi cum officia vel magni
doloribus qui repudiandae
vero nisi sit
quos veniam quod sed accusamus veritatis error",
    "id": 10,
    "title": "optio molestias id quia eum",
    "userId": 1,
  },
]
`;

exports[`TransferState should add state to page 2 1`] = `undefined`;

exports[`TransferState should add state to page 2 2`] = `
Object {
  "address": Object {
    "city": "Gwenborough",
    "geo": Object {
      "lat": "-37.3159",
      "lng": "81.1496",
    },
    "street": "Kulas Light",
    "suite": "Apt. 556",
    "zipcode": "92998-3874",
  },
  "company": Object {
    "bs": "harness real-time e-markets",
    "catchPhrase": "Multi-layered client-server neural-net",
    "name": "Romaguera-Crona",
  },
  "email": "Sincere@april.biz",
  "id": 1,
  "name": "Leanne Graham",
  "phone": "1-770-736-8031 x56442",
  "username": "Bret",
  "website": "hildegard.org",
}
`;
